### DISCUSSION TITLE
Leave a comment here if you are a user

### DISCUSSION THREAD

**Discussion by laszlocph** [woodpecker-ci/woodpecker#2149] [Show and tell]
Please comment on this issue if you are using Woodpecker.

It would mean a lot. Would keep the spirit, and help convincing new users.

Add the nature of your usage as well: company / private, build volume per week

**Comment by paweldaborowski** [comment]
In our legal services company we are using Woodpecker as a main CI/CD platform to great satisfaction, with hundreds of builds per day. It's tailor cut to our needs, simple and flexible. Friendly license is a great plus. 
Thoroughly recommended!

**Comment by Nikoos** [comment]
I am starting to learn CICD (from a sysadmin POV), and after testing Drone, I move to Woodpecker. I was able to link woodpecker to my internal gitea server (in https).

**Comment by JensHeinrich** [comment]
I'm currently still mostly testing and not yet fully using it

**Comment by PeterAlabaster** [comment]
I've been testing / building a knowledge base on different CI platforms for my company over the past few months intermittently. Tried a myriad of different platforms but some were overly complex and restrictive. We are a smallish company who develop solutions on the odoo framework mostly, but also do other things.

I landed on drone but the gray licensing lead me here. We've started the process of migrating active projects over to a woodpecker system we set up, after verifying it suits most of our needs (and after you addressed some of our issues!)

We have a use case where we have many projects which utilize mostly similar pipeline steps, so the custom plugin functionality is amazing for managing the pipeline scripts centrally, and managing which steps get executed on a per-project basis using the yaml file. This way, if our release process changes slightly in the future, we can just update that plugin and have the change propagate to all projects if we want.

Being able to override the default clone step with a plugin is great, as most of our repositories are a mix of private gitea --> private/public github submodules, and we weren't willing to change everything to use https instead of git@, so we just inject the private key (whose pubkey is attached to a robot user on both vcs systems) and clone recursively.

We've opted for building / testing on PR opening, and additional steps of releasing / notifying a new image upon merge, so we will probably have ~20-40 builds a week once all our projects are on there.

Just want to say thanks for your work on this!

**Comment by laszlocph** [comment]
:hugs: 

The system design is the work of the original Drone author Brad Rydzewski.

I just try to not mess it up too much and keep the 0.8 line alive, for the very same reasons.. it's simple, it's good enough, and I use it a lot :)

**Comment by taz77** [comment]
I am using Woodpecker for a POC/Startup I am working on. I have it running out of my house on my personal Docker Swarm. I wanted a Container first CI. I have used a lot of different CI's on my own and at my day job - Jenkins, Bamboo, Jenkins, Gitlab CI. Woodpecker, and Drone by extension, was what I wanted.

**Comment by 3nprob** [comment]
Just starting out evaluating here. Really appreciate the spirit of the project <3

**Comment by cschildknecht-fielddata** [comment]
I'm just in evaluation of container based CI/CD. I came here from Drone project, since OSS is not viable for Kubernetes.
Thanks for your work!

**Comment by ChewingBever** [comment]
Hi everyone,
I'm a computer science student from Belgium and I'd love to set this up on my servers, but I can't seem to find any information about how to set this up using Gitea. From the docs I was able to derive the following:

```
- 'DRONE_GITEA=true'
- 'DRONE_GITEA_CLIENT=${GITEA_CLIENT}'
- 'DRONE_GITEA_SECRET=${GITEA_SECRET}'
```

Where `GITEA_CLIENT` is the URL to my Gitea instance and `GITEA_SECRET` my OAuth secret. Is this correct? I based this off [this page](https://woodpecker.laszlo.cloud/server-setup/) as there was no official link to any Gitea documentation.

I'd really appreciate any help, as I'm really excited to start using this!

Edit:
> Add the nature of your usage as well: company / private, build volume per week

Kinda forgot about this. I would personally use this for all the projects me and my friends are making, so my build volume would probably be somewhere around 20-30 builds per week (my servers aren't that powerful so I wouldn't be able to go any higher).

**Comment by inverse** [comment]
We adopted Woodpecker moving from a Drone 0.8 setup. Migration was pretty painless and our engineers are happier with proper multi-pipeline support.

We have 400+ repos and are doing around  4k builds a week.

Thanks for all your hard work :)

**Comment by laszlocph** [comment]
@ChewingBever Gitea doesn't use OAuth in Woodpecker, perhaps this will help you in your setup: https://github.com/laszlocph/woodpecker/issues/130#issuecomment-675982171

**Comment by laszlocph** [comment]
> We adopted Woodpecker moving from a Drone 0.8 setup. Migration was pretty painless and our engineers are happier with proper multi-pipeline support.
> 
> We have 400+ repos and are doing around 4k builds a week.

@inverse This is amazing. Thanks for sharing!
If you or others have any input on the direction of the project, I'm interested: https://github.com/laszlocph/woodpecker/issues/165#issuecomment-792546528

I see that Gitea is an interest for many. A Gitea push with OAuth would worth the effort..

**Comment by jtackaberry** [comment]
> I see that Gitea is an interest for many. A Gitea push with OAuth would worth the effort..

Definitely of interest for me.  I'm a current Drone user (just a small personal setup) but since they got picked up by Harness they fell into a blackhole and at some point even disabled issue tracking on GitHub -- so much for that "vibrant, opensource community" (apologies for sounding bitter) -- so I've been keeping a watchful eye on Woodpecker's Gitea support for a while.

**Comment by helmut72** [comment]
I use Gitea as my Git repository for my own Dockerfiles and docker-compose.yml to manage my private server. My environment is a good timesaver, but there are some boring and error prone tasks left. To fill this gap I installed Woodpecker. All in all Woodpecker looks good, but I found a strange behavior: https://github.com/laszlocph/woodpecker/issues/130#issuecomment-841794909

Edit: BTW, I'm using Woodpecker just for one day. For now it's more playing with it and to learn more about CI.

**Comment by ChanceNCounter** [comment]
I too use Gitea, self-hosted. I haven't had a need for local CI for a while, but the last time I used it, my Jenkins instance had gone kaput, so I spun up Woodpecker a few months ago just to have it ready.

I finally needed it today. I'm currently trying to suss out which parts of the Drone API are and aren't working as I expect, but otherwise 💯 I'm on board.

**Comment by catdevnull** [comment]
We are trying out Woodpecker for [Sutty](https://sutty.nl), a worker cooperative. We just started so I don't have a good measurement of how many builds we use just yet.

**Comment by ivictbor** [comment]
Hey @laszlocph, thanks a lot for the hard fork and maintenance! 

I am the founder of https://devforth.io/ software development company, we loved Drone CI because of it's simplicity and containerization idea, we used it for our clients a lot and recommended it a lot. At that time, as a contribution to the great tool and great idea, we created several blog posts and a set of guides which popularized DroneCI, however as it turned out this was a mistake.

Once it was unexpectedly sold, harness made unnecessarily huge price, and of course, we don't want to popularize Drone anymore. Thanks to you we are now moving to woodpecker. A lot of projects and a lot of work in progress.... But I think it should be capable because what we need is only basic configs, secrets + slack plugin. If this movement will be successful we will destroy all our guides about Drone and adjust them for woodpecker.

**Comment by 6543** [comment]
@ivictbor that would be awesome, we are currently restructuring woodpecker to make it modern and easier to maintain. So next version will be breaking a lot with legacy ... just that you konw ... (esp. have a look at #218)

---
*just a side-note, for the sake of funding ;) woodpecker is now on [OpenCollective](https://opencollective.com/woodpecker-ci)*

**Comment by ivictbor** [comment]
@6543 Thanks for the info, the first build was handled successfully! We deployed https://tracklify.com using Woodpecker! Our DevOps engineer said the setup was pretty smooth and straightforward. The first small donation should also come soon!

**Comment by hairlesshobo** [comment]
I have recently started using woodpecker internally for my own personal open source and private development. I have it tied to my internal gitea server and is working great. I tried drone first but couldn't get it to build at all.. woodpecker was easy to setup and runs great!

**Comment by reivilibre** [comment]
I've been using Woodpecker together with my personal Gitea on my Raspberry Pi 4 and I've been liking it! The docs could do with some love perhaps and I have a few thoughts on how I'd like to make the build syntax better, but no real complaints from me :). Thanks

**Comment by BracketJohn** [comment]
Hey!

We're a software agency based in Cologne, Germany and are planning to move to woodpecker from drone. Same reason as @ivictbor: No more open source, huge price tags and no more development progress in drone ci. We have around 40-50 repos and around 100-200 builds per week. 

Thank you so much for forking off of drone and maintaining this project! ❤️

**Comment by wzzrd** [comment]
Just started using this today and it is the bee’s knees! Awesome work, and thank you so much! Love it!

I’m using woodpecker for my personal projects for now, running partially on my NAS and partially on some Raspberry Pi’s. So much better than anything else out there!

**Comment by EsmailELBoBDev2** [comment]
I started to use it since gitlab stopped free pull mirrors for public repos. use it on gitea

it's nice but syntax is still has learning curve for me

**Comment by builderdotnet** [comment]
Estoy evaluando usar woodpecker para proyectos NET (Full y core) pero tenia la duda si soportaba CD.

**Comment by 6543** [comment]
support of CD: I would say yes ... with plugins

**Comment by S1SYPHOS** [comment]
I'm using Woodpecker over at codeberg and just love it. So far nothing that couldn't be solved by looking stuff up in the docs - keep up the good work!

**Comment by Eragonfr** [comment]
I'm a user of woodpecker too. Just using it on my personal gitea instance. It I run the server on my RockPro64 board, and I'm using my computer as the agent (My board will never support to run more than one build at a time anyway).

**Comment by simmstein** [comment]
I use woodpecker every single day.

**Comment by imolein** [comment]
I used Drone CI for years, but after the v1.0 I got pretty annoyed with it, especially that they ripped out useful feature to sell them as enterprise features. Then they were bought by harness which finally made me search for an alternative and I found woodpecker-ci and from one day to another suddently l had all my missing features back (and more new ones).

Long story short, I happily use woodpecker daily, so thanks to the whole team/community :)

**Comment by gapodo** [comment]
I use Woodpecker for my private projects (docker, K8s and local backend), as well as within the Forgejo community (where I'm part of the team maintaining our CI), which has migrated the Drone pipelines from our upstream Gitea over to Woodpecker and hasn't looked back once.

While I'm still forced to use gitlab at work, I've started to convince the infra team I'm a part of to consider using woodpecker for some of our use cases, as it's easier to handle, more portable, gives us better performance and doesn't have us relying on gitlab for everything.

tl;dr Works great for "home" use, communities and starts to make an entry at my work place due to the benefits over the "large / well known" solutions.

**Comment by wiktor-k** [comment]
I'm using Woodpecker hosted by Codeberg and it's really great! Of course there are rough edges here and there but having such a complex system *and* it being open-source is really mind-blowing!

Thanks for your hard-work and have a nice day. :wave:

**Comment by dessalines** [comment]
We're starting to move from drone -> woodpecker for [Lemmy](https://github.com/lemmynet/lemmy), including our server, and UI. It's great to have a responsive CI community again, because drone felt dead.

**Comment by foxyseta** [comment]
I am currently using Woodpecker on Codeberg as a private. It's great! I just look up what I need in the docs and everything works fine.

**Comment by tionis** [comment]
I am currently using woodpecker as my main CI and backend of some simple git-apps. 
It's been working pretty much flawless, and I considered switching to Gitea's actions, but I like woodpeckers simple container based design more, and it just works.
So thanks for all the good work!

**Comment by 6543** [comment]
@gedw99 thanks for the comment ... for now the workaround would be the local-agent-backend ... but it it's some "docker container" woodpecker can already use it ...

**Comment by ghost** [comment]
I'm using Woodpecker with a self hosted server on Hetzner as well. So far I managed to cover every CI step I had in mind. Woodpecker runs builds, tests, SAST, publishes releases, gpg signs checksum files and so on. 
Writing the pipeline feels very intuitive after a short introduction. This is the first CI solution I'm using btw.
I'd also like to highlight the docs. I think they are very well written.

**Comment by rigrig** [comment]
I've been using it for a while now for my private projects, works great for my needs.

**Comment by philipkozeny** [comment]
We've also migrated away from Gitlab Enterprise edition to Gitea and Woodpecker and our team is really happy about that change. I would say that our setup is now running even better than before.

**Comment by vyskocilm** [comment]
Use it on my small private k8s server as a part of my migration projects to codeberg.

**Comment by Lite5h4dow** [comment]
Im a user, looking to move from drone CI. Gitness is a hot turd. using Gitea & Drone rn, cant seem to solve a Service acc issue with woodpecker for my main usecase

**Reply by 6543** [reply]
feel free to ask via matrix (chat) how to solve it ... almost all things should be possible and if not, it would be interesting to know to us :)

**Comment by nightm4re94** [comment]
I've been happily using Woodpecker in the context of Codeberg CI for some smaller GameDev-related projects for around 1,5 years now. What a lovely tool to have!

**Comment by rdehuyss** [comment]
We from https://github.com/jobrunr/jobrunr just switched from Drone to Woodpecker CI. Thanks again for maintaining this excellent fork! You guys are awesome 🙌 🥰!

**Comment by locnide** [comment]
I finally installed woodpecker to replace drone CI 1.x for my forgejo instance.
I’m so glad that I can write CI pipelines again ! *(was a bit reluctant to write drone pipelines knowing that I had to migrate)*
Thanks for maintaining it !!
